### ISSUE TITLE
Better boss label system.

### ISSUE THREAD

**Issue by FrankenMonkey** [AdventureLookup/AdventureLookup#137]
I am proposing that the boss be labeled by name like Strahd then have a little tag beside the name that would say vampire or beholder or orc. This just allows more detail in that section as well as providing the necessary search tags like you said in the video.

**Comment by MattColville** [comment]
You know what, I think that's how it's actually supposed to work! Like, right now!

Right now you can add a person, like Strahd, and it will prompt you to also tell it what type of monster Strahd is.

That seems good to me. I don't _mind_ an entry like "Nightscale" as long as it also says "(Black Dragon)."

The problem is the "Type" field for Boss Monster is A: currently validated which means you have to pick from a list, and B: badly populated I think. It's using some predefined list of types which I already see lots of problem with.

But I don't think we need Type to be a field, people should just be able to type in "Zuggtmoy (Demon Lord)" and it should work. It should return that adventure if people search for Zuggtmoy, Demon, or Demon Lord, and that means it's working.

Plus that way curators can easily edit stuff.

**Comment by harkenn** [comment]
The problem is ensuring that type is present and not left off (most likely), written incorrectly, or written correctly with typos. BBEGs are treated just the same as monsters (which is why they have a Type). In a separate ticket, you had made clear that the primary use case is to recognize a specific "type" of monster as the Villain for an adventure. Your point above, @MattColville seems to contract what you had mentioned in that other ticket (I'd have to go find it, but I'm pretty sure its one that's still open).

**Comment by MattColville** [comment]
I think the issue is what we mean when we say "Type." I mean species or maybe race/class. I don't mean "the category listed in the MM entry."

I know folks hate it when I say "No one is going to..." but no one is going to search for "Adventures with Humanoids as villains" because that is way too broad. It includes many wildly different monsters. But they WILL search for "Adventures with Goblins as bad guys."

_In general_ I don't think the name is important for obvious reasons. But I think allowing folks to say "Lareth the Beautiful" as long as they then also go on to say "(Human Cleric)" is fine. And there are some notable, named monsters out there.

So really I just think letting people put their own parenthetical label should be fine?

**Comment by harkenn** [comment]
I understand, sure that would be fine. Can you speak to whether or not that's possible right now?

For humanoid, should we remove that as a type and replace it with detailed versions of the most likely uses of humanoid? It would increase the length of the type list significantly.

**Comment by MattColville** [comment]
I'm not even sure we need Type for boss monster. I think it asks too much of the user. "Hey is this bad guy a humanoid or a monstrous humanoid." A lot of people won't know the difference and those terms didn't exist for like 25 years and won't be in the source material for a lot of adventures.

Instead, folks should just list name and, in parentheses, the kind. Black Dragon, Human Cleric.

**Comment by harkenn** [comment]
If there is no type, then the primary use case you described `... no one is going to search for "Adventures with Humanoids as villains" because that is way too broad. It includes many wildly different monsters. But they WILL search for "Adventures with Goblins as bad guys."` is handled poorly.

When a user creates a boss monster for an adventure and mistypes the Boss's type, leaves it off, etc. Then that adventure is lost in that respect. I see lots of bad things coming out of excluding monster type from the data points for villains. As for users not being able to know/find that information, is that any more difficult to find than - say - publisher, or the writers? 

A villains' type is going to be in the adventure. Every DM knows what "type" is and why its significant (unless they're very very new to D&D and they'll have lots of issues with using AL). Type is an integral combat mechanic and has been for many, many years.

**Comment by MattColville** [comment]
>A villains' type is going to be in the adventure.

Nope. Type is something D&D3 deployed and it has not been used consistently across editions. It wasn't around for AD&D or second edition, or basic, or original D&D. 

And sort of the point of Adventure Lookup is that older adventures are still good and useful.

As long as folks enter "Bogun Redcap (goblin)" then I think we've handled the vast majority of use cases. We can expect users to search for "adventures with demons as bad guys" but expecting them to know that, actually, demons are "Type: outsider" is jargon and beyond the scope of what we're trying to do.

**Comment by harkenn** [comment]
I don't understand, why would a demon be an outsider? Aren't Demons a fiend? Or is that a difference only employed in more recent editions?

**Comment by loudobjectprincess** [comment]
Demons are fiends, fiends are outsiders, therefore demons are outsiders.

**Comment by MattColville** [comment]
Yeah this thread is actually now a good example of why the type field is suboptimal.

However, I think maybe we just need a better type field. "Type" is a game mechanic in some editions of D&D.

Instead what folks would like is to show "all adventures with demons and devils as the bad guys." Ditto Undead, or Elf.

So gimme some time and I'll come up with a better type list.

Then the convention will be Name (species) [type]
